mas-foundation/src/main/java/com/ca/mas/foundation/MASDeviceMetadata.java

```diff
@@ -0,0 +1,186 @@
+            }
+        });
+    }
+
+    public static void getAttributes(final MASCallback<JSONArray> callback){
+
+        MASRequest request = null;
+        try {
+            request = new MASRequest.MASRequestBuilder(new URI(ENDPOINT_PATH))
+                    .header(HEADER_KEY, MAG_IDENTIFIER)
+                    .get()
+                    .build();
+        } catch (URISyntaxException e) {
+            Callback.onError(callback, e);
+        }
+
+        MAS.invoke(request, new MASCallback<MASResponse<JSONArray>>() {
+            @Override
+            public void onSuccess(MASResponse<JSONArray> response) {
+                if (checkResponse(response)) {
+                   callback.onSuccess(response.getBody().getContent());
+                }
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                Callback.onError(callback, e);
+            }
+        });
+    }
+
+    public static void deleteAttribute(String attr, final MASCallback<Void> callback) {
+        MASRequest request = null;
+        String route = ENDPOINT_PATH + "/" +attr;
+
+        try {
+            request = new MASRequest.MASRequestBuilder(new URI(route))
+                    .header(HEADER_KEY, MAG_IDENTIFIER)
+                    .delete(MASRequestBody.stringBody(attr))
+                    .build();
+        } catch (URISyntaxException e) {
+            callback.onError(e);
+        }
+
+        MAS.invoke(request, new MASCallback<MASResponse<String>>() {
+            @Override
+            public void onSuccess(MASResponse<String> response) {
+                if (checkResponse(response) && response.getBody().getContent().equalsIgnoreCase("removed")) {
+                    callback.onSuccess(null);
+                }
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                Callback.onError(callback, e);
+            }
+        });
+    }
+
+    public static void deleteAttributes(final MASCallback<Void> callback) {
+        MASRequest request = null;
+        try {
+            request = new MASRequest.MASRequestBuilder(new URI(ENDPOINT_PATH))
+                    .header(HEADER_KEY, MAG_IDENTIFIER)
+                    .delete(MASRequestBody.stringBody(""))
+                    .build();
+        } catch (URISyntaxException e) {
+            Callback.onError(callback, e);
+        }
+
+        MAS.invoke(request, new MASCallback<MASResponse<String>>() {
+            @Override
+            public void onSuccess(MASResponse<String> response) {
+                if (checkResponse(response) && response.getBody().getContent().equalsIgnoreCase("removed")) {
+                    callback.onSuccess(null);
+                }
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                Callback.onError(callback, e);
+            }
+        });
+    }
+
+    private static boolean checkResponse(MASResponse response) {
```

Comment:
You don't have to checkResponse range, the SDK will invoke onSuccess within the HTTP Success range and onError for others.